coral-service/src/main/java/com/linkedin/coral/coralservice/controller/TranslationController.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2022 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.coralservice.controller;
+
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.annotation.Profile;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import static com.linkedin.coral.coralservice.utils.CoralProvider.*;
+import static com.linkedin.coral.coralservice.utils.TranslationUtils.*;
+
+
+/**
+ * Class defining the REST endpoints to translate queries using Coral.
+ */
+@RestController
+@Service
+@Profile({ "remoteMetastore", "default" })
+public class TranslationController implements ApplicationListener<ContextRefreshedEvent> {
+
+  @Override
+  public void onApplicationEvent(ContextRefreshedEvent event) {
+    // runs after the Spring context has been initialized
+    try {
+      initHiveMetastoreClient();
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  @GetMapping("/translate")
+  public ResponseEntity translate(@RequestParam String query, @RequestParam String fromLanguage,
```

Comment:
should we define fromLanguage and toLanguage as enum? or cast them to enum?
It will probably make validation and the if-else logic simpler.

Reply:
I think using enum still needs the same amount of code for if-else and validation blocks?
If not, could you give me an example?